diff --git a/tests/test_api.py b/tests/test_api.py
index ee11a8d..4472b85 100644
--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -427,0 +428,8 @@ class TestLowLevel:
+
+
+def test_overlay_enable_async(env):
+    assert not env.is_async
+    assert not env.overlay().is_async
+    env_async = env.overlay(enable_async=True)
+    assert env_async.is_async
+    assert not env_async.overlay(enable_async=False).is_async
