diff --git a/src/click/utils.py b/src/click/utils.py
index 223d6b8..67922b8 100644
--- a/src/click/utils.py
+++ b/src/click/utils.py
@@ -235 +235 @@ def echo(
-    message: t.Any | None = None,
+    message: object = None,
@@ -290,5 +290,7 @@ def echo(
-    # Convert non bytes/text into the native string type.
-    if message is not None and not isinstance(message, (str, bytes, bytearray)):
-        out: str | bytes | bytearray | None = str(message)
-    else:
-        out = message
+    match message:
+        case str() | bytes() | bytearray():
+            out = message
+        case None:
+            out = ""
+        case _:
+            out = str(message)
@@ -297 +298,0 @@ def echo(
-        out = out or ""
@@ -313 +313,0 @@ def echo(
-
