diff --git a/src/jinja2/environment.py b/src/jinja2/environment.py
index f062e40..8219717 100644
--- a/src/jinja2/environment.py
+++ b/src/jinja2/environment.py
@@ -409 +409 @@ class Environment:
-        enable_async: bool = False,
+        enable_async: bool = missing,
@@ -421,0 +422,3 @@ class Environment:
+        .. versionchanged:: 3.1.5
+            ``enable_async`` is applied correctly.
+
@@ -423 +426 @@ class Environment:
-            Added the ``newline_sequence``,, ``keep_trailing_newline``,
+            Added the ``newline_sequence``, ``keep_trailing_newline``,
