diff --git a/src/urllib3/connectionpool.py b/src/urllib3/connectionpool.py
index 7e6f54c8..427e5dfb 100644
--- a/src/urllib3/connectionpool.py
+++ b/src/urllib3/connectionpool.py
@@ -585 +585 @@ class HTTPConnectionPool(ConnectionPool, RequestMethods):
-        if self.port and not port:
+        if self.port is not None and port is None:
@@ -587 +587 @@ class HTTPConnectionPool(ConnectionPool, RequestMethods):
-        elif not self.port and port == port_by_scheme.get(scheme):
+        elif self.port is None and port == port_by_scheme.get(scheme):
diff --git a/src/urllib3/poolmanager.py b/src/urllib3/poolmanager.py
index b7248d21..935712bf 100644
--- a/src/urllib3/poolmanager.py
+++ b/src/urllib3/poolmanager.py
@@ -312 +312 @@ class PoolManager(RequestMethods):
-        if not port:
+        if port is None:
@@ -599 +599 @@ class ProxyManager(PoolManager):
-        if not proxy.port:
+        if proxy.port is None:
diff --git a/src/urllib3/util/url.py b/src/urllib3/util/url.py
index 6d0c66a3..36529382 100644
--- a/src/urllib3/util/url.py
+++ b/src/urllib3/util/url.py
@@ -159 +159 @@ class Url(
-        if self.port:
+        if self.port is not None:
